### PR TITLE
fix(inquiry): inquiry q/a

### PR DESCRIPTION
**Pull request by dzucconi**
Along with the recent Palette updates, this PR handles the majority of [the Q/A items](https://www.notion.so/artsy/Artwork-Inquiry-Form-QA-7102f7c454a54003b5263ed13fa8a54e). I'm going to take a look at GDPR separately (which AFAIK we don't address in the current implementation either).

### CODE REVIEW

File: src/desktop/components/location_search/index.styl [hunk 1]
```diff
@@ -16,41 +16,3 @@
...
-  &.pac-item-selected
-    border-color transparent
-    background-color #000
-    > .pac-item-query
-      color #fff
-
-.pac-item,
-.pac-item-query
-  garamond()
-  font-size default-garamond-font-size
```
Comment (dzucconi) [inline src/desktop/components/location_search/index.styl:56]: Old instances of this will just use the default Google styling, which is fine. The new component has styling that matches the rest of our autocompletes.

File: src/v2/Components/Inquiry/Hooks/useArtworkInquiryRequest.ts [hunk 2]
```diff
@@ -1,55 +1,56 @@
...
 > & {
-  relayEnvironment?: Environment
   artworkID: string
   message: string
   contactGallery?: boolean | null
 }
 
 export const useArtworkInquiryRequest = () => {
-  const { relayEnvironment: defaultRelayEnvironment } = useSystemContext()
+  const { relayEnvironment } = useInquiryContext()
```
Comment (damassi) [inline src/v2/Components/Inquiry/Hooks/useArtworkInquiryRequest.ts:18]: Why do we pull the `relayEnvironment` from inquiry context rather than system context?

Comment (dzucconi) [reply src/v2/Components/Inquiry/Hooks/useArtworkInquiryRequest.ts:18]: This gets set in login/sign up. We authenticate the environment and the one in system context is still logged out.

File: src/v2/Components/Inquiry/Hooks/useInquiryContext.tsx [hunk 3]
```diff
@@ -116,6 +122,15 @@ const InquiryProvider: React.FC<InquiryProviderProps> = ({
     onDone: onClose,
   })
 
+  const { relayEnvironment: defaultRelayEnvironment } = useSystemContext()
+
+  const relayEnvironment = useRef(defaultRelayEnvironment!)
+
+  const setRelayEnvironment = (updatedEnvironment: Environment) => {
```
Comment (damassi) [inline src/v2/Components/Inquiry/Hooks/useInquiryContext.tsx:129]: A comment here describing why this is necessary might be helpful for later

File: src/v2/Components/Inquiry/Views/InquiryLogin.tsx [hunk 4]
```diff
@@ -51,10 +57,15 @@ export const InquiryLogin: React.FC = () => {
     try {
       const { user } = await login({ email: inquiry.email!, ...state })
 
+      const relayEnvironment = createRelaySSREnvironment({ user })
```
Comment (damassi) [inline src/v2/Components/Inquiry/Views/InquiryLogin.tsx:60]: Why is this necessary? (Code comment here would be good since we don't have to do this anywhere else in the app)

**Review by damassi — APPROVED** [review]
Just a couple non-blocking questions / suggestions. Looks good 🍏

File: src/v2/Components/Inquiry/Hooks/useArtworkInquiryRequest.ts (hunk 2 above)
Comment (damassi) [reply src/v2/Components/Inquiry/Hooks/useArtworkInquiryRequest.ts:18]: Another thought is: can we move this behavior up to the system context itself and add the function there? (This would resolve other questions in PR, as code comments there would be unnecessary.) Thinking through it, currently when we log out we do a hard reload, but in the future that doesn't have to be so, and in the meantime we could at least provide a mechanism to refresh relay's environment from where it's typically accessed.

Comment (dzucconi) [reply src/v2/Components/Inquiry/Hooks/useArtworkInquiryRequest.ts:18]: Explored that a bit; it won't work because we'd need to return the environment ref rather than the environment itself. Meaning it'd lead to a much larger refactor. It's a good idea to figure this out but I'm just going to add some comments here for now.

Comment (damassi) [reply src/v2/Components/Inquiry/Hooks/useArtworkInquiryRequest.ts:18]: Np 👍

File: src/v2/Components/Inquiry/Hooks/useInquiryContext.tsx [hunk 5]
```diff
@@ -126,6 +127,11 @@ const InquiryProvider: React.FC<InquiryProviderProps> = ({
 
   const relayEnvironment = useRef(defaultRelayEnvironment!)
 
+  /**
+   * In this flow we go from a logged out state to a logged in/signed up state
+   * where we have to execute mutations like sending the inquiry, saving your
+   * information, etc. We store the Relay environment in a ref then update it here.
+   */
```
Comment (damassi) [inline src/v2/Components/Inquiry/Hooks/useInquiryContext.tsx:134]: 👍
